### DISCUSSION TITLE
v4 UI Issues Error WIX0200

### DISCUSSION THREAD

**Discussion by mwileczka** [wixtoolset/issues#7372] [Questions] [answered]
I read https://github.com/orgs/wixtoolset/discussions/7115 and think this is still a problem, or related. I followed the help there and it didn't solve the problem.

I tried converting an old sample project with rc4 and had a number of issues, so I figured I'd wait until official 4 came out. I tried making a basic sample project with VS and Heatwave and it's still not working. I upgraded everything to 4 from 4rc4. Brand new blank "MSI Package" builds just fine. When I try to add a UI, it starts giving me errors.

The error is "Error	WIX0200	The Package element contains an unhandled extension element 'WixUI'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/ui' namespace has been provided."

I added WixToolset.Util.wixext via NuGet as project reference.
```
<Project Sdk="WixToolset.Sdk/4.0.0">
  <ItemGroup>
    <PackageReference Include="WixToolset.Netfx.wixext" Version="4.0.0" />
    <PackageReference Include="WixToolset.Util.wixext" Version="4.0.0" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\XYZ Client\XYZ Client.csproj" />
  </ItemGroup>
</Project>
```

This is at top of Package.wxs
`<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:ui="http://wixtoolset.org/schemas/v4/wxs/ui">`

I added this to <Package>
```
    <ui:WixUI Id="WixUI_Mondo" />
    <ui:WixUI Id="WixUI_ErrorProgressText" />
```
and it fails to build.

**Comment by barnson** [comment, marked as answer]
https://wixtoolset.org/docs/fourthree/faqs/#referencing-the-standard-wixui-dialog-sets

**Comment by mwileczka** [comment]
My apologies, I had included the wrong package. I needed the UI not Util package. Not sure if I just kept reading that wrong, old info, or weird conversion.

**Comment by robmen** [comment]
I don't want to single out this issue but for everyone's reference, this...

> ...so I figured I'd wait until official 4 came out.

...is _exactly_ the wrong thing to do.

If you want something to work for free (i.e. you do not pay for a support contract that guarantees bug fixes), then you need to stay on top of all the previews and RCs and work with the maintainers by opening good bug reports and participating in discussions for scenarios vital to your work.

This time commitment to upgrade regularly, investigate then report issues you see, and generally participate in discussions around the project is the price you pay for free open source. If you choose not to participate, then _at best_ you are waiting for the next cycle to see your scenario work. At worst, your scenario was already discarded.

Oh, and even with that participation, there are no guarantees. If you aren't a paying customer, you're at the whims of the interest of the maintainers. The more active you are, the better they know you, the better the chance they care about your scenario.

Again, my comment is less about this particular issue and just something everyone keeps in mind generally.

**Reply by mwileczka** [reply]
Sorry, bad choice of words, touchy subject from watching the video a few days ago.

I started looking into Wix with some sample projects a number of months ago in order to get rid of an inherited InstallShield project :vomiting_face: . Unfortunately I got side tracked until last week when I found out about v4. I tried to convert my sample project but ran into some issues that I had found documented due to converter issues (references), but also just things in my sample that the converted didn't handle (spaces in project names for one). I also found Wix 4 Dojo videos about trying to stop old habits. So I figured I'd just wait until the official release and start from scratch based on new schema and practices.

I think I missed a few things in the documentation. I know this is a transition period and there just isn't a lot of samples or videos for v4 yet, so I'm not trying to get on anyone's case, as you said, this is a community project. However, I feel obligated to comment that there is a need for some sample v4 projects. If they exist somewhere on GitHub or the main site, they are just hard to find. Knowing what best practices are (or new intended practices) has been a challenge.

A search utility in the documentation would go a long way in helping newbies as myself find something.

I do plan on getting involved in the community more once I get my feet wet. Thank you for your hard work.
